imagetest/test_suites/packagevalidation/package_test.go

```diff
@@ -110,39 +109,61 @@ func TestGuestPackages(t *testing.T) {
 		}
 	}
 
-	// What packages to insure are not installed
-	excludePkgs := []string{}
-
-	// What packages to check for and excude for diffrent images
-	// Redhat like is default
-	requiredPkgs := []string{"google-guest-agent", "google-osconfig-agent"}
-	requiredPkgs = append(requiredPkgs, "google-compute-engine", "gce-disk-expand", "google-cloud-cli")
-	requiredPkgs = append(requiredPkgs, "google-compute-engine-oslogin")
-
-	// We haven't published a rocky-linux-8-optimized for arm with google-cloud-cli, so check google-cloud-sdk instead.
-	if strings.Contains(image, "rocky-linux-8-optimized-gcp-arm64") {
-		requiredPkgs = removeFromArray(requiredPkgs, "google-cloud-cli")
-		requiredPkgs = append(requiredPkgs, "google-cloud-sdk")
-	}
-
-	if strings.Contains(image, "sles") || strings.Contains(image, "suse") {
-		requiredPkgs = removeFromArray(requiredPkgs, "google-cloud-cli")
-		requiredPkgs = removeFromArray(requiredPkgs, "google-compute-engine")
-		requiredPkgs = removeFromArray(requiredPkgs, "google-compute-engine-oslogin")
-		requiredPkgs = removeFromArray(requiredPkgs, "gce-disk-expand")
-		requiredPkgs = append(requiredPkgs, "google-guest-configs") // SLES name for 'google-compute-engine' package.
-		requiredPkgs = append(requiredPkgs, "google-guest-oslogin")
-	}
-	if strings.Contains(image, "centos-7") {
-		requiredPkgs = append(requiredPkgs, "epel-release")
-	}
-	if strings.Contains(image, "debian") {
-		requiredPkgs = append(requiredPkgs, "haveged", "net-tools")
-		requiredPkgs = append(requiredPkgs, "google-cloud-packages-archive-keyring", "isc-dhcp-client")
-		excludePkgs = append(excludePkgs, "cloud-initramfs-growroot")
-	}
-	if strings.Contains(image, "ubuntu") {
-		requiredPkgs = removeFromArray(requiredPkgs, "gce-disk-expand")
+	pkgs = []*osPackage{
+		&osPackage{
+			name: "google-guest-agent",
+		},
+		&osPackage{
+			name: "google-osconfig-agent",
+		},
+		&osPackage{
+			name:       "google-compute-engine",
+			imagesSkip: []string{"sles", "suse"},
+		},
+		&osPackage{
+			name:   "google-guest-configs",
+			images: []string{"sles", "suse"},
+		},
+		&osPackage{
+			name:   "google-guest-oslogin",
+			images: []string{"sles", "suse"},
+		},
+		&osPackage{
+			name:       "gce-disk-expand",
+			imagesSkip: []string{"sles", "suse", "ubuntu"},
+		},
+		&osPackage{
+			name:         "google-cloud-cli",
+			alternatives: []string{"google-cloud-sdk"},
+		},
+		&osPackage{
+			name:       "google-compute-engine-oslogin",
+			imagesSkip: []string{"sles", "suse"},
+		},
+		&osPackage{
+			name:   "epel-release",
+			images: []string{"centos-7"},
```

Comment:
There is a bug in the existing logic, we should check this on `rhel-7` too.

Reply:
done. thank you.